src/main/java/de/rabea/gui/JavaFXUi.java

```diff
@@ -18,7 +18,9 @@ public void setGameRunner(GameRunner gameRunner) {
 
     @Override
     public void displayBoard(Board board, Player player) {
-        viewUpdater.showBoard((GuiPlayer) player, board, gameRunner, false);
+        if (!player.hasMove()) {
+            viewUpdater.showBoard((GuiPlayer) player, board, gameRunner, false);
```

Comment:
Hmm.. an unchecked cast. How do you make sure 'player' is a GuiPlayer?
This type-cast looks like a code smell worth investigating...